packages/components/dropdowns/dropdown-menu/src/dropdown-menu.spec.tsx

```diff
@@ -0,0 +1,67 @@
+import SecondaryButton from '@commercetools-uikit/secondary-button';
+
+import { act, screen, render } from '../../../../../test/test-utils';
+import DropdownMenu from './dropdown-menu';
+
+describe('DropdownMenu', () => {
+  it('should render list menu', async () => {
+    const firstOptionOnClick = jest.fn();
+    const secondOptionOnClick = jest.fn();
+
+    await render(
+      <DropdownMenu
+        menuType="list"
+        triggerElement={<SecondaryButton label="Trigger" />}
+      >
+        <DropdownMenu.ListMenuItem onClick={firstOptionOnClick}>
+          Option 1
+        </DropdownMenu.ListMenuItem>
+        <DropdownMenu.ListMenuItem onClick={secondOptionOnClick} isDisabled>
+          Option 2
+        </DropdownMenu.ListMenuItem>
+      </DropdownMenu>
+    );
+
+    // Dropdown shold be closed by default
+    expect(screen.queryByText('Option 1')).not.toBeVisible();
+
+    // Open the dropdown
+    screen.getByLabelText('Trigger').click();
+    expect(await screen.findByText('Option 1')).toBeVisible();
+    expect(screen.getByText('Option 2')).toBeVisible();
+
+    // Clicking in the disabled options should do nothing
+    screen.getByText('Option 2').click();
+    expect(secondOptionOnClick).toHaveBeenCalledTimes(0);
+    expect(await screen.findByText('Option 1')).toBeVisible();
+
+    // Clicking in the enabled option should close the dropdown
+    screen.getByText('Option 1').click();
+    expect(firstOptionOnClick).toHaveBeenCalledTimes(1);
+    expect(await screen.findByText('Option 1')).not.toBeVisible();
+  });
+
+  it('should render the default menu', async () => {
+    await render(
+      <>
+        <h2>Header</h2>
+        <DropdownMenu triggerElement={<SecondaryButton label="Trigger" />}>
+          <div>Content</div>
+        </DropdownMenu>
+      </>
+    );
+
+    // Dropdown shold be closed by default
+    expect(screen.queryByText('Content')).not.toBeVisible();
+
+    // Open the dropdown
+    screen.getByLabelText('Trigger').click();
+    expect(await screen.findByText('Content')).toBeVisible();
+
+    // Clicking outside the dropdown should close it
+    act(() => {
+      screen.getByText('Header').click();
+    });
```

Comment:
I think `act` is needed here, due to [this effect](https://github.com/commercetools/ui-kit/blob/842892e4a33ece64c64dd741b532ce4126164941/packages/components/dropdowns/dropdown-menu/src/dropdown-menu.tsx#L88-L93) which changes component state after click. 
Just as a side note, I noticed though that changing this line to `screen.getByRole('button', { name: 'Trigger' }).click();` makes all work without `act`

Reply:
Thanks @kark 

However, I still don't understand as that effect does not change the component state but just removes an event listener from the `window` object.

Reply:
Sorry for not being precise, the effect doesn't change the state directly, the event handler it adds does. It's hard for me to reason about it in details but I guess reasoning aligns with what was mentioned [here](https://kentcdodds.com/blog/fix-the-not-wrapped-in-act-warning#the-dreaded-act-warning:~:text=If%20you%27re%20still-,experiencing,-the%20act%20warning)